application/app/models/upload_file.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+class UploadFile < ApplicationDiskRecord
+  include ActiveModel::Model
+  include LoggingCommon
+
+  ATTRIBUTES = %w[id project_id type file_location filename status size creation_date start_date end_date metadata].freeze
+
+  attr_accessor *ATTRIBUTES
+
+  validates_presence_of :id, :project_id, :type, :file_location, :filename, :status, :size
+  validates :size, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, allow_nil: true
```

Comment:
If we add the class grouping upload files, then `type`, `project_id`, and `metadata` will move to that class.
This will be a much simple class

Reply:
will do on the next PR